src/buffer.h

```diff
@@ -154,4 +175,50 @@ mpd_buffer_consume(struct mpd_buffer *buffer, size_t nbytes)
 	buffer->read += nbytes;
 }
 
+/**
+ * Makes at least min_size bytes available for writing data.
+ * In other words, mpd_buffer_room() will be >= min_size if called after this
+ * function.
+ */
+static inline bool
+mpd_buffer_make_room(struct mpd_buffer *buffer, size_t min_avail_len)
+{
+	size_t newsize;
+
+	if (mpd_buffer_room(buffer) >= min_avail_len)
+		return true;
+
+	newsize = buffer->data_len * 2;
+	while (newsize < min_avail_len)
+		newsize *= 2;
+
+	/* empty buffer */
+	if (mpd_buffer_room(buffer) == buffer->data_len) {
+		free(buffer->data);
+		buffer->data = malloc(newsize);
+		if (buffer->data == NULL)
+			return false;
+		memset(buffer->data, 0, newsize);
```

Comment:
Why `memset()`?

Reply:
For the same reason it is used in `mpd_buffer_init`: initialize data to a known state